test/unit.js

```diff
@@ -0,0 +1,273 @@
+            const webComponentContent = reactRootContainer.parentNode;
+
+            expect(ReactTestUtils.isDOMComponent(componentContents)).to.equal(true, 'component contents');
+            expect(isReactRoot(reactRoot)).to.equal(true, 'react root');
+            expect(reactRootContainer.tagName).to.equal('DIV', 'react root container');
+            expect(webComponentContent.className).to.equal('content', 'web component content');
+        });
+
+        it('> DOM element): properties are passed to the wrapped React component', function () {
+            var renderedApp = renderApp(
+                React.createClass({
+                    render: function () {
+                        return <ReactComponent>
+                            <em id="user-inserted-contents">{this.props.text}</em>
+                        </ReactComponent>;
+                    }
+                })
+            );
+
+            renderedApp.setProps({text: 'initial text'});
+            expect(document.getElementById('user-inserted-contents').textContent).to.equal('initial text');
+            renderedApp.setProps({text: 'updated text'});
+            expect(document.getElementById('user-inserted-contents').textContent).to.equal('updated text');
+        });
+
+        it('> React component): properties flow down into the child React component', function () {
+            const ContainedReactComponent = React.createClass({
+                render: function () {
+                    return <em id="user-inserted-contents">{this.props.text}</em>
+                }
+            });
+
+            var renderedApp = renderApp(
+                React.createClass({
+                    render: function () {
+                        return <ReactComponent>
+                            <ContainedReactComponent text={this.props.text}/>
+                        </ReactComponent>;
+                    }
+                })
+            );
+
+            renderedApp.setProps({text: 'initial text'});
+            expect(document.getElementById('user-inserted-contents').textContent).to.equal('initial text');
+            renderedApp.setProps({text: 'updated text'});
+            expect(document.getElementById('user-inserted-contents').textContent).to.equal('updated text');
+        });
+
+        it('> DOM element): events triggered on the DOM element can be handled on click', function () {
+            var clicked = false;
+            function handleClick () {
+                clicked = true;
+            }
+
+            renderApp(
+                React.createClass({
+                    render: function () {
+                        return <ReactComponent>
+                                    <em onClick={handleClick} id="user-inserted-contents">Click me</em>
+                            </ReactComponent>;
+                    }
+                })
+            );
+
+            ReactTestUtils.Simulate.click(document.getElementById('user-inserted-contents'));
```

Comment:
You could also do `skate.emit(document.getElementById('user-inserted-contents'), 'click')`.